src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs

```diff
@@ -1640,14 +1596,10 @@ private void DisplayCancelWaitMessage()
             {
                 _taskLoggingContext.LogWarningFromText(null, warningCode, helpKeyword, new BuildEventFileInfo(_taskLocation), message);
             }
-            catch (InternalErrorException) // BUGBUG, should never catch this
```

Comment:
Do you think it's ok to catch this now? Or should we leave the comment that we shouldn't be but are "temporarily"?

Reply:
This didn't really sound like it was a bug to me. Sure, it would be better if we never threw that exception, and we could accomplish something like that by having something to check whether we were cancelled before throwing, but having a catch feels equivalent to me as long as there aren't other ways internal exceptions can come through here. Perhaps it would be slightly better if we had `catch (InternalException) when (!_taskLoggingContext.IsValid && weWerentCancelled)`, but for a try/catch around a single logging statement, that feels like overkill.